### ISSUE TITLE
Minifying output

### ISSUE THREAD

**Issue by visj** [DamianEdwards/RazorSlices#57]
In line with the "interesting to explore" section, have you considered adding an option to minify the html output passed to the razor engine? Basically add the "trim insignificant whitespace" option that nowadays exist for Blazor projects.

I tried to give it a go myself, but it seems today that there is no logic for how to modify the generated Razor code. What was your intended idea how to implement that inside the source generator?

I would love to help out if I can. I found your repo through the TechEmpower benchmarks implementation. They actually minified the html manually themselves for that benchmark.

**Comment by DamianEdwards** [comment]
Yeah source generators can't be chained or combined so there's very little that can be done from the context of this library without massive gymnastics and hackery, e.g. MSBuild-based code gen of minimized *.cshtml* files that then become the input to the Razor source generator. The Razor Compiler itself is the one doing the HTML minification for *.razor* files because in the world of Blazor they're much more concerned with the particulars of the HTML syntax and structure (because of the way their engine does diff-based rendering) whereas the *.cshtml* dialect of Razor doesn't go down the code-path with that logic (as far as I know, @phil-allen-msft or @SteveSandersonMS might know for sure).

**Comment by ivanjx** [comment]
currently i use this and it works perfectly https://www.nuget.org/packages/WebMarkupMin.Core#readme-body-tab
be mindful to always set `RemoveOptionalEndTags` to `false`.
